EtlPipelineBlobTrigger/course_stats.py

```diff
@@ -730,72 +767,70 @@ def need_unavailable(self, xml_elem):
             return True
         return False
 
-    def get_unavailable_reason_subj(self, sbj_key):
-        if sbj_key:
-            return self.get_english_sbj_label(sbj_key)
-        return "this subject"
+    def get_aggs_for_code(self, unavail_reason_code):
+        return self.unavail_reason_english["data"][unavail_reason_code].keys()
+
+    def get_unavailable(self, elem):
+        unavailable = {}
+        subj_key = elem.get(self.xml_subj_key)
+        agg = elem[self.xml_agg_key] if self.has_data(elem) else None
+        unavail_reason_code = elem[self.xml_unavail_reason_key]
+        validate_unavailable_reason_code(unavail_reason_code)
+
+        unavailable["code"] = int(unavail_reason_code)
+        unavailable["reason_english"] = self.get_unavailable_reason_str(
+            unavail_reason_code, subj_key, agg, elem
+        )
+        unavailable["reason_welsh"] = self.get_unavailable_reason_str(
+            unavail_reason_code, subj_key, agg, elem, welsh=True
+        )
+        return unavailable
 
     def get_unavailable_reason_str(
-        self, unavail_reason_code, subj_key, agg, xml_elem
+        self, unavail_reason_code, subj_key, agg, xml_elem, welsh=False
     ):
         validate_unavailable_reason_code(unavail_reason_code)
+        if welsh:
+            unavail_reason_lookup = self.unavail_reason_welsh
+        else:
+            unavail_reason_lookup = self.unavail_reason_english
 
         if not self.has_data(xml_elem):
-            reason_str = self.unavail_reason["no-data"][unavail_reason_code]
+            reason_str = unavail_reason_lookup["no-data"][unavail_reason_code]
             return unicodedata.normalize("NFKD", reason_str)
 
-        validate_agg(unavail_reason_code, agg, self.unavail_reason)
-        partial_reason_str = self.unavail_reason["data"][unavail_reason_code][
-            agg
-        ]
+        validate_agg(unavail_reason_code, agg, unavail_reason_lookup)
+        partial_reason_str = unavail_reason_lookup["data"][
+            unavail_reason_code
+        ][agg]
         partial_reason_str = unicodedata.normalize("NFKD", partial_reason_str)
-        subj = self.get_unavailable_reason_subj(subj_key)
+        if welsh:
+            subj = self.get_unavailable_reason_subj_welsh(subj_key)
+            subj = ""  # TODO Remove this line when translations avaialble
```

Comment:
Not sure why you would need this, just return empty string from lookup?

Reply:
Never mind I see why you've done this now from looking at the welsh lookup table.